go-controller/pkg/ovn/pods_test.go

```diff
@@ -897,6 +897,206 @@ var _ = ginkgo.Describe("OVN Pod Operations", func() {
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
 
+		ginkgo.It("correctly stops retrying adding a pod after failing n times", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespace1 := *newNamespace("namespace1")
+				podTest := newTPod(
+					"node1",
+					"10.128.1.0/24",
+					"10.128.1.2",
+					"10.128.1.1",
+					"myPod",
+					"10.128.1.3",
+					"0a:58:0a:80:01:03",
+					namespace1.Name,
+				)
+				pod := newPod(podTest.namespace, podTest.podName,
+					podTest.nodeName, podTest.podIP)
+
+				key, err := getResourceKey(factory.PodType, pod)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				fakeOvn.startWithDBSetup(initialDB,
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							namespace1,
+						},
+					},
+					&v1.PodList{
+						Items: []v1.Pod{*pod},
+					},
+				)
+
+				podTest.populateLogicalSwitchCache(fakeOvn, getLogicalSwitchUUID(fakeOvn.controller.nbClient, "node1"))
+				fakeOvn.controller.WatchNamespaces()
+				fakeOvn.asf.ExpectAddressSetWithIPs(podTest.namespace, []string{podTest.podIP})
+				gomega.Eventually(fakeOvn.controller.nbClient).Should(
+					libovsdbtest.HaveData(getExpectedDataPodsAndSwitches([]testPod{}, []string{"node1"})...))
+
+				// inject transient problem, nbdb is down
+				fakeOvn.controller.nbClient.Close()
+				gomega.Eventually(func() bool {
+					return fakeOvn.controller.nbClient.Connected()
+				}).Should(gomega.BeFalse())
+
+				// trigger pod add, which will fail with "context deadline exceeded: while awaiting reconnection"
+				fakeOvn.controller.WatchPods()
+				// sleep long enough for TransactWithRetry to fail, causing pod add to fail
+				time.Sleep(ovstypes.OVSDBTimeout + time.Second)
+
+				// wait until retry entry appears
+				gomega.Eventually(func() *retryObjEntry {
+					return fakeOvn.controller.retryPods.getObjRetryEntry(key)
+				}).ShouldNot(gomega.BeNil())
+
+				// check that the retry entry is marked for creation
+				retryEntry := fakeOvn.controller.retryPods.getObjRetryEntry(key)
+				gomega.Expect(retryEntry).ToNot(gomega.BeNil())
+				gomega.Expect(retryEntry.newObj).ToNot(gomega.BeNil())
+				gomega.Expect(retryEntry.failedRetries).To(gomega.Equal(uint8(0)))
+				// set failedRetries to maxRetries-1, trigger a retry (which will fail due to nbdb being down)
+				// and verify that failedRetries is now equal to maxRetries
+				fakeOvn.controller.retryPods.setFailedRetriesCounter(key, maxRetries-1)
+				fakeOvn.controller.retryPods.requestRetryObjs()
+				gomega.Eventually(func() uint8 {
+					entry := fakeOvn.controller.retryPods.getObjRetryEntry(key)
+					return entry.failedRetries
+				}).Should(gomega.Equal(uint8(maxRetries))) // no more retries are allowed
```

Comment:
so added retry entry by bringing down NBDB then u manually inc the threshold so we are at the max but u didn't really ensure from that point on we can't retry 
maybe u should init retry to max then force the error and make sure the error presists because we can't retry ?

Reply:
Hmm... are you saying that I should make sure that the code won't trigger a retry on its own, while I modify the retry entry? The trigger will happen 30 seconds later, so I thought we have an ample margin before that happens.  Or did I miss your point? :)

Reply:
what I am saying initial retry threshold to the max with ur test api then force error as usual make sure retry entry is there and verify we are at the max now reconnect NBDB the error stick because we won't do retry anymore because we hit the limit.
then u can manually edit the counter with ur api to bring the counter down so iterate can do retry and fix the condition, basically I wanted to see the impact on retry functionality than just check the counter is incrementing , make sense ?

Reply:
on reading the test again it really does what I exactly wanted so scratch this comment